src/libinjector/linux/linux_utils.c

```diff
@@ -0,0 +1,298 @@
+ *                                                                         *
+ * If you have received a written license agreement or contract for        *
+ * Covered Software stating terms other than these, you may choose to use  *
+ * and redistribute Covered Software under those terms instead of these.   *
+ *                                                                         *
+ * Source is provided to this software because we believe users have a     *
+ * right to know exactly what a program is going to do before they run it. *
+ * This also allows you to audit the software for security holes.          *
+ *                                                                         *
+ * Source code also allows you to port DRAKVUF to new platforms, fix bugs, *
+ * and add new features.  You are highly encouraged to submit your changes *
+ * on https://github.com/tklengyel/drakvuf, or by other methods.           *
+ * By sending these changes, it is understood (unless you specify          *
+ * otherwise) that you are offering unlimited, non-exclusive right to      *
+ * reuse, modify, and relicense the code.  DRAKVUF will always be          *
+ * available Open Source, but this is important because the inability to   *
+ * relicense code has caused devastating problems for other Free Software  *
+ * projects (such as KDE and NASM).                                        *
+ * To specify special license conditions of your contributions, just say   *
+ * so when you send them.                                                  *
+ *                                                                         *
+ * This program is distributed in the hope that it will be useful, but     *
+ * WITHOUT ANY WARRANTY; without even the implied warranty of              *
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the DRAKVUF   *
+ * license file for more details (it's in a COPYING file included with     *
+ * DRAKVUF, and also available from                                        *
+ * https://github.com/tklengyel/drakvuf/COPYING)                           *
+ *                                                                         *
+ * This file was created by Manorit Chawdhry.                              *
+ * It is distributed as part of DRAKVUF under the same license             *
+ ***************************************************************************/
+
+
+#define _GNU_SOURCE // required for memmem
+#include "linux_injector.h"
+#include "linux_debug.h"
+#include <sys/mman.h>
+#include <fcntl.h>
+
+addr_t find_vdso(drakvuf_t drakvuf, drakvuf_trap_info_t* info)
+{
+    addr_t process_base = drakvuf_get_current_process(drakvuf, info);
+    PRINT_DEBUG("Process base: %lx\n", process_base);
+
+    ACCESS_CONTEXT(ctx,
+        .translate_mechanism = VMI_TM_PROCESS_PID,
+        .pid = info->proc_data.pid
+    );
+
+    addr_t addr = 0;
+    size_t offset = 0;
+
+    vmi_instance_t vmi = drakvuf_lock_and_get_vmi(drakvuf);
```

Comment:
Please, rewrite vmi releasing using goto statements:
https://www.kernel.org/doc/html/latest/process/coding-style.html#centralized-exiting-of-functions

Reply:
Sure, will do it. Thanks for a good coding style guide, I was really searching for one but couldn't find it

Reply:
Done